### DISCUSSION TITLE
LOXILB NAT gateway

### DISCUSSION THREAD

**Discussion by jamroks** [loxilb-io/loxilb#617] [Q&A] [answered]
Hi all , 

I was wondering, is it possible to use Loxilb as nat gateway ( nat44 and nat64), in the contexte of AWS vpc, for kubernetes workload and also **non kubernetes** workload apps at the same time ? 

Aws native natgateway are too expensive so i was looking for a way to have loxilb  with the following requirement: 

All this should be serve from the same loxilb instance 
- loxilb + kube-loxilb Service Loadbalancer  : this is already possible i know 
- loxilb nat44 : **non kubernetes**  ??
- loxilb nat64 : **non kubernetes**  ??

**Comment by nik-netlox** [comment]
Hi @jamroks , Do you need nat44/nat64 for specific non-kubernetes service or you want to use loxilb as a generic NAT gateway?

**Reply by jamroks** [reply]
it's kind of both at the same times , let me put it like this, imagine for a second that you have private subnet which has : 
- kubernetes cluster with apps deployed on it , and are made publicly accessible through loxilb + Kube-loxilb 👍 

now you also have some legacy workloads deployed on VM , that run in the same private subnet and need nat44/nat64 

-  is there a way to use the previous loxilb node (that already provide service of type loadbalancer to kubernetes workload),  to make legacy workload outbound traffic routed through loxilb node ? loxilb node would acting or serving both purpose.

The first one being Service of type loadbalancer  and the second is NAT gateway for outbound traffic from private subnet

**Comment by nik-netlox** [comment]
Currently, we do not support SNAT feature but it seems like your requirement can be accomplished with fullnat mode. You can manually configure a lb rule with any VIP as service IP and VM IP as Endpoint IP. loxilb will replace the source IP with the VIP and forwards the traffic to your VM and it will handle the reverse traffic accordingly. Please read more about fullnat mode [here](https://loxilb-io.github.io/loxilbdocs/nat/).

**Reply by nik-netlox** [reply, marked as answer]
We also support service with custom endpoint where you can create a load balancer service with external endpoints. Please check [here](https://github.com/loxilb-io/loxilbdocs/blob/main/docs/ext-ep.md) and let us know if any of the suggested methods works for you.

**Reply by jamroks** [reply]
really cool thank you, i'll try
